config/src/test/java/tech/pegasys/pantheon/config/GenesisConfigFileTest.java

```diff
@@ -214,6 +215,59 @@ public void acceptComments() {
     // Unfortunately there is no good (non-flakey) way to assert logs.
   }
 
+  @Test
+  public void testOverridePresent() {
+    final GenesisConfigFile config = GenesisConfigFile.development();
+    final int bigBlock = 999_999_999;
+    final String bigBlockString = Integer.toString(bigBlock);
+    final Map<String, String> override = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
```

Comment:
Can we add some tests that use a case-sensitive `Map` with various casing strategies for the keys to verify that values are replaced in a case-insensitive way?

Reply:
Done